src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py

```diff
@@ -337,11 +337,25 @@ def trash_incarnation_dir(self):
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
 
-    def remove(self, retainsnaps=False):
+    @staticmethod
+    def safe_to_remove_subvolume_clone(internal_cleanup):
+        # Both the STATE_FAILED and STATE_CANCELED are handled by 'handle_clone_failed' in the state
+        # machine which removes the entry from the index. Hence, it's safe to removed clone with
+        # force option for both.
+        acceptable_rm_clone_states = [SubvolumeStates.STATE_COMPLETE, SubvolumeStates.STATE_CANCELED,
+                                      SubvolumeStates.STATE_FAILED]
+        if not internal_cleanup and self.state not in acceptable_rm_clone_states:
```

Comment:
Umm.. Wouldn't it better to pass the state as an argument to this?

Reply:
Yeah that's a bug while moving into static method. I will fix it.